### PR TITLE
Update Golang to 1.20 version

### PR DESCRIPTION
**Pull request by anikin-aa**
### Database name

All of the databases type

# Pull request description

Updating of golang version to 1.20, just to keep it up to date

### Describe what this PR fix
// problem is ...

### Please provide steps to reproduce (if it's a bug)
// it can really help

### Please add config and wal-g stdout/stderr logs for debug purpose

also you can use WALG_LOG_LEVEL=DEVEL for logs collecting
<details><summary>If you can, provide logs</summary>
<p>
```bash
any logs here
```
</p>
</details>

### CODE REVIEW

**Review by serprex — APPROVED** [review]

**Comment by anikin-aa** [comment]
hmm, interesting for some reason github actions set golang ver to 1.2:

```
Setup go version spec 1.2
Attempting to download 1.2...
matching 1.2...
Not found in manifest.  Falling back to download directly from Go
Install from dist
Acquiring go1.2.2 from https://storage.googleapis.com/golang/go1.2.2.linux-amd64.tar.gz
Extracting Go...
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/f58b1e98-a993-434c-9adf-24ce428ab2e7 -f /home/runner/work/_temp/8760a75c-d083-42ff-9[23](https://github.com/wal-g/wal-g/actions/runs/5265448579/jobs/9534097632?pr=1492#step:5:24)3-f0f306573d53
Successfully extracted go to /home/runner/work/_temp/f58b1e98-a993-434c-9adf-[24](https://github.com/wal-g/wal-g/actions/runs/5265448579/jobs/9534097632?pr=1492#step:5:25)ce4[28](https://github.com/wal-g/wal-g/actions/runs/5265448579/jobs/9534097632?pr=1492#step:5:29)ab2e7
Adding to the cache ...
Successfully cached go to /opt/hostedtoolcache/go/1.2.2/x64
Added go to the path
Setting GOROOT for Go version < 1.9
Successfully set up Go version 1.2
/opt/hostedtoolcache/go/1.2.2/x64/bin/go env GOMODCACHE
/opt/hostedtoolcache/go/1.2.2/x64/bin/go env GOCACHE


Warning: Restore cache failed: Could not get cache folder paths.
go version go1.2.2 linux/amd64
```

checking

**Comment by anikin-aa** [comment]
Oh, seems need to specify minor version too https://github.com/actions/setup-go/issues/326

File: .github/workflows/dockertests.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ env:
   IMAGE_UBUNTU: wal-g/ubuntu
   IMAGE_GOLANG: wal-g/golang
   IMAGES_CACHE_KEY: docker-images-${{ github.sha }}
-  GO_VERSION: 1.18
+  GO_VERSION: 1.20
```
[the remark this answers is not in the document]
Comment (anikin-aa) [reply .github/workflows/dockertests.yml]: Yes, seems like. Thanks for pointing out. Changes are committed.

**Review by serprex — APPROVED** [review]

**Comment by anikin-aa** [comment]
Seems like all of the tests are fine, except MariaDB, that fixed in scope of https://github.com/wal-g/wal-g/pull/1495

File: .golangci.yml [hunk 2]
```diff
@@ -82,7 +82,7 @@ linters:
     - whitespace
 
 run:
-  go: 1.18
+  go: 1.20
```
Comment (serprex) [inline .golangci.yml:85]:
```suggestion
  go: "1.20"
```

**Pull request merged**